pkg/storaged/client.js

```diff
@@ -689,7 +689,8 @@
 
         function start() {
             var buf = "";
-            python.spawn([ inotify_py, vdo_monitor_py ], [ ], { superuser: "try", err: "message" })
+            cockpit.spawn([ "python", "--", "-" ], { superuser: "try", err: "message" })
+                .input(inotify_py + vdo_monitor_py)
```

Comment:
So on an OS which does not have Python 2 installed, monitoring will just silently fail, but one could still control VDO operations, right? The bit that I don't really like is that this will now not work with Python 3 at all, even if at some point that vdomgmt module will appear for Python 3.

Perhaps `python.spawn()` can become more clever to not use `which`, but just actually try and run the program with python3 first, and if it fails, run it through python2? Then the former should fail with an `ImportError`. Of course this needs to be validated to our other usages of `python.spawn()` (i. e. that non-zero exit code of the code that we run is not an expected answer). If that  happens, we could introduce a new `python.check_spawn()`, resembling Python's `subprocess.check_exec()`, which does that?

Reply:
> So on an OS which does not have Python 2 installed, monitoring will just silently fail, but one could still control VDO operations, right?

This whole things makes a lot of shaky assumptions, and one of them is that /usr/bin/vdo is written in Python 2 and comes with a private vdomgmt module.  We have blessing to use that module (the alternative being to directly read /etc/vdocfg.yml and files in /sys instead).  So, as long as /usr/bin/vdo is there, we can assume that python is also there and works. However, we should figure out a better way forward.

(There might be a proper vdomgmt module for Python 3 at some point, or a API in UDisks2, maybe based on new udev properties.)

I wouldn't want to make our normal, clean use of Python any more complicated because of the sins of our VDO code.

I plan out the VDO cleanup properly.

Reply:
OK. So this will unbreak VDO on Fedora 27 without introducing a potential regression for now. So let's go with that for now and revisit that once there is a better API. Thanks!